internal/fleet/agent_policy_resource.go

```diff
@@ -264,6 +270,11 @@ func resourceAgentPolicyRead(ctx context.Context, d *schema.ResourceData, meta i
 }
 
 func resourceAgentPolicyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if d.Get("skip_destroy").(bool) {
+		tflog.Debug(ctx, "Skipping destroy of Agent Policy", map[string]interface{}{"policy_id": d.Id()})
```

Comment:
IMO this should be at a higher log level so it's clear to users that we're not destroying this policy during apply.